### PR TITLE
feat(http): set the default Accept header

### LINKED ISSUE #6354: Http requests to content-negotiated API endpoints fail on Firefox
http://plnkr.co/edit/FamsVu1O3MxShrQIyMCy?p=preview

We investigated this and found that the default "Accept" header is different between Firefox and Chrome. 

Chrome defaults to: `"*/*"`
Firefox defaults to: `"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"`

Angular 1 handled this for us. With Angular 2, we're required to modify the headers to request JSON or accept the server default.

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Fixes https://github.com/angular/angular/issues/6354

### CODE REVIEW

File: modules/@angular/http/test/backends/xhr_backend_spec.ts [hunk 1]
```diff
@@ -237,6 +237,29 @@ export function main() {
         expect(setRequestHeaderSpy).toHaveBeenCalledWith('X-Multi', 'a,b');
       });
 
+      it('should attach default Accept header', () => {
+        const headers = new Headers({'Content-Type': 'text/xml', 'Breaking-Bad': '<3'});
+
+        const base = new BaseRequestOptions();
+        const connection = new XHRConnection(
+            new Request(base.merge(new RequestOptions({headers: headers}))), new MockBrowserXHR());
```
Comment (vicb) [inline modules/@angular/http/test/backends/xhr_backend_spec.ts]: `{headers}`

**Comment by vicb** [comment]
Please address the few comments and reword the commit message `fix(http):...`

**Comment by DzmitryShylovich** [comment]
done

**Comment by vicb** [comment]
Thanks !

**Pull request merged**